### DISCUSSION TITLE
How to make schema an entire optional

### DISCUSSION THREAD

**Discussion by shyakadavis** [open-circle/valibot#483] [Q&A] [answered]
Hi 👋 

Say I have a `post_schema` with `title`&`content` fields, all required.

How could I create an extended schema called `update_post_schema` from `post_schema` whose fields are `ALL` optional, but, if a user does provide input to the field, then the validation should apply/kick in regardless of the optional behaviour?

Can't seem to find a concrete `CRUD` example from the docs.

Thanks.

**Comment by fabian-hiller** [comment, marked as answer]
Have a look at `partial`: https://valibot.dev/api/partial/

**Reply by shyakadavis** [reply]
Hello.

Thanks a lot. I had found `optional` but it generated an optional schema but not optional entries, and couldn't find this.

Cheers.
